src/luxos/version.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+import sys
+import types
+from pathlib import Path
+from typing import Literal, overload
+
+__version__ = ""
+__hash__ = ""
+
+
+@overload
+def get_version(
+    modules: list[types.ModuleType] | None, as_string: Literal[True]
+) -> str: ...
+
+
+@overload
+def get_version(
+    modules: list[types.ModuleType] | None, as_string: Literal[False]
+) -> dict[str, str]: ...
+
+
+@overload
+def get_version(
+    modules: list[types.ModuleType] | None, as_string: bool
+) -> str | dict[str, str]: ...
+
+
+def get_version(
+    modules: list[types.ModuleType] | None = None, as_string: bool = False
```

Comment:
maybe I'm missing context here but most of the calls to `get_version` seem to be `as_string = True` wouldn't be better to make it the default behavior instead of False?

just a comment not a change request

Reply:
Yeah, I should've done that. In a later change.